### PR TITLE
multicall: bubble up revert reason

### LINKED ISSUE #175: [posm]: Bubble up reverts
### Component

Position Manager

### Describe the suggested feature and problem it solves.

Like v4-core we should bubble up reverts. Needed when we use multicall in posm

### Describe the desired implementation.

_No response_

### Describe alternatives.

_No response_

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by saucepoint**
## Related Issue
Closes #175 

## Description of changes

Modify Multicall.sol to bubble up custom reverts with arguments

### CODE REVIEW

File: src/base/Multicall.sol [hunk 1]
```diff
@@ -20,7 +26,7 @@ abstract contract Multicall is IMulticall {
                     }
                 }
                 // Next 5 lines from https://ethereum.stackexchange.com/a/83577
-                if (result.length < 68) revert();
+                if (result.length < 68) CallFailed.selector.bubbleUpAndRevertWith();
```
Comment (snreynolds) [inline src/base/Multicall.sol]: not sure about this 68 length check.. but why wouldnt we want to bubble up if > 68 ?

File: src/base/Multicall.sol [hunk 2]
```diff
@@ -1,30 +1,28 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 pragma solidity ^0.8.19;
 
+import {CustomRevert} from "@uniswap/v4-core/src/libraries/CustomRevert.sol";
+
 import {IMulticall} from "../interfaces/IMulticall.sol";
 
+import "forge-std/console2.sol";
```
Comment (snreynolds) [inline src/base/Multicall.sol:8]: remove?

**Review by snreynolds — COMMENTED** [review]
i think once you merge the other bubble reverts into this PR would be nice to test with those Wrapped reverts to make sure we're surfacing those properly in the multicall

File: test/Multicall.t.sol [hunk 3]
```diff
@@ -106,4 +107,112 @@ contract MulticallTest is Test {
...
+        calls[0] = abi.encodeWithSelector(MockMulticall(multicall).revertWithBytes.selector, data);
+
+        // revert is caught
+        vm.expectRevert(abi.encodeWithSelector(MockMulticall.ErrorBytes.selector, data));
+        multicall.multicall(calls);
+
+        // confirm expected length of the revert
+        try multicall.revertWithBytes(data) {}
+        catch (bytes memory reason) {
+            // errors with 0 bytes are by default 64 bytes of data (length & pointer?) + 4 bytes of selector
```
Comment (snreynolds) [inline test/Multicall.t.sol:183]: errors with 0 bytes of data outside the selector? thats super interesting.. didnt know that i guess that explains the 68 number

**Review by snreynolds — APPROVED** [review]
noice looks good to me

File: src/base/Multicall.sol (hunk 1 above)
Comment (hensha256) [reply src/base/Multicall.sol]: because in old-solidity errors couldnt be less than 68 bytes, so it meant there was no error to bubble

**Pull request merged**